### PR TITLE
Run mypy on our test code

### PR DESCRIPTION
**Pull request by bmw**
I wanted to this as part of my testing cleanup to help ensure we don't break things while refactoring. I'm planning to start doing this for the Certbot repo soon.

Most of this PR was generated automagically using [MonkeyType](https://github.com/Instagram/MonkeyType). This is what was done in the first commit where I

1. [Set up and activated a dev environment](https://github.com/certbot/josepy/blob/master/CONTRIBUTING.md#configure-a-development-environment).
2. Ran:
```
pip install monkeytype
cd tests
monkeytype run -m pytest
for m in $(monkeytype list-modules | grep -v josepy); do monkeytype apply $m; done
```
The follow up commits were made manually except for the last one which was generated using `poetry run isort src tests`.

### CODE REVIEW

**Review by alexzorin — CHANGES_REQUESTED** [review]
Sorry, I should have reviewed this first. Merge conflicts.

File: tests/jwk_test.py [hunk 1]
```diff
@@ -17,24 +25,36 @@
...
         from josepy.jwk import JWKRSA
         self.assertRaises(errors.Error, JWKRSA.load, DSA_PEM)
 
 
+class JWKSubclassTest(Protocol):
+    from josepy.jwk import JWK
+    jwk: JWK
+    thumbprint: bytes
+
+    def assertEqual(self, *unused_args: Any) -> None:
```
Comment (alexzorin) [inline tests/jwk_test.py:42]: I suppose this should be removed once rebased with master due to removing `unittest` in e6cea11dea17fa6936576accf878c9a4862fa3b1.

**Review by alexzorin — COMMENTED** [review]
Other than the merge conflicts, this looks good.

**Comment by bmw** [comment]
No worries about the merge conflicts. To explain how I resolved them to try and help with the rereview, I ran `git merge master -X ours`, redid the steps I described in https://github.com/certbot/josepy/pull/154, and then fixed a couple new minor problems in the last commits.

File: tests/jwk_test.py (hunk 1 above)
Comment (bmw) [reply tests/jwk_test.py]: Nice catch. I definitely would have missed this.

**Review by alexzorin — APPROVED** [review]

**Pull request merged**